#157731617: Relation between WorkoutLog and WorkoutSession

King-Benx opened the pull request:
### What does this PR do?
Creates a relation between the workout log and workout session
#### Description of Task to be completed?
Previously, wger only linked the workout session and workout log based on date and this presented some problems in certain areas since it was not able to follow foreign keys with the ORM is some location.

This PR creates a foreign key of the Workout Log.

#### How should this be manually tested?
1. Follow the readme.rst procedures on how to set up the application locally 
2. Make sure you run the migrations command
3. The database should contain a field for log_id in the workout session

#157731617

dondrzzy commented on wger/core/static/js/wger-core.js:
```diff
@@ -119,6 +119,20 @@ function setProfileField(field, newValue) {
+  } else {
+    $('#id_sets').val(1);
+    $('#id_sets_value').html($('#id_sets').val());
+    updateAllExerciseFormset();
```
Dont repeat yourself here. Move the bottom 2 lines of code in both if-else conditions and place it at the bottom. They will always be called in both cases.

dondrzzy commented on wger/core/static/js/wger-core.js:
```diff
@@ -697,4 +716,73 @@ $(document).ready(function () {
       '/' + token;
     window.location.href = targetUrl;
   });
+
```
These are not your changes. When you git pull --rebase. Endeavour to squash and always have your commit at the top.

dondrzzy reviewed (commented):
These are not your changes. When you git pull --rebase. Endeavor to squash and always have your commit at the top.

hariclerry commented on wger/manager/views/workout.py:
```diff
@@ -87,6 +87,8 @@ def view(request, pk):
     canonical = workout.canonical_representation
     uid, token = make_token(user)
 
+    # for day in workout.canonical_representation['day_list']:
```
Please remove this commented line of code.

emugaya reviewed (dismissed):
Fix your commit message

King-Benx committed:
feat(relation btn log and session): Add relation between log and session

Originally, workout logs and the workout session are only linked
through the date (for sessions this value is unique and enforced
by the DB). This presents in some places a problem, since it's
not possible to just follow a foreign key relationship with the
ORM like in other places.

This commit creates a log relation between the WorkoutLog and
WorkoutSession

[Delivers #157731617]
